### PR TITLE
fix fallback for scaling blas number with array

### PR DESCRIPTION
**Pull request by KristofferC**
Also adds a fast path for special case of zero and one.

https://github.com/JuliaLang/LinearAlgebra.jl/issues/351

### CODE REVIEW

File: base/linalg/dense.jl [hunk 1]
```diff
@@ -18,6 +20,8 @@ function scale!{T<:BlasFloat}(X::Array{T}, s::T)
     X
 end
 
+scale!{T<:BlasFloat}(s::T, X::Array{T}) = scale!(X, s)
```
Comment (tkelman) [inline base/linalg/dense.jl]: why does this method exist? really strange to be sometimes mutating the second input, but usually the first

Comment (ViralBShah) [inline base/linalg/dense.jl]: Perhaps to be symmetric like multiplication?

Comment (andreasnoack) [inline base/linalg/dense.jl]: Because multiplication is not always commutative.

Comment (KristofferC) [inline base/linalg/dense.jl]: I also thought it was weird. I guess it is there for non commutative numbers in the general case.

Comment (tkelman) [inline base/linalg/dense.jl]: mutating methods need to be consistent about what they're mutating.

Comment (andreasnoack) [inline base/linalg/dense.jl]: No they don't. We don't agree here and for those of us who use this method, it doesn't cause trouble.

Comment (Keno) [inline base/linalg/dense.jl]: It's mutating the array, whichever argument that is.

Comment (tkelman) [inline base/linalg/dense.jl]: Added to #16772. Being explicit with 3 arguments seems like a much more predictable API when you're writing generic code that has to deal with various combinations of input types.

**Comment by KristofferC** [comment]
Apple seemed to got stuck on something unrelated. Since this should be fairly uncontroversial I'm gonna click the scary button for the first time. :tada:

**Pull request merged**